src/libpgmoneta/restore.c

```diff
@@ -698,13 +777,27 @@ pgmoneta_combine_backups(int server, char* label, char* base, char* input_dir, c
       goto error;
    }
 
-   pgmoneta_art_destroy(prior_backups);
+   if (number_of_workers > 0)
+   {
+      pgmoneta_workers_wait(workers);
+      if (!workers->outcome)
+      {
+         goto error;
+      }
+      pgmoneta_workers_destroy(workers);
+   }
+
+   pgmoneta_art_destroy(backups);
    pgmoneta_deque_iterator_destroy(iter);
    free(server_dir);
    return 0;
 
 error:
-   pgmoneta_art_destroy(prior_backups);
+   if (number_of_workers > 0)
```

Comment:
destroy should work on `NULL`, no ?

Reply:
Maybe, I'm just copying over. It's written this way in other places.